src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,481 @@
+                                 Math.min(text.length(), LIMIT_CHARACTER - 1)));
+
+                MTHUBTranslateRequest mttr = new MTHUBTranslateRequest(
+                        mthubKey, normaliseCode(sLang), normaliseCode(tLang),
+                        segments);
+
+                String json = OM.writeValueAsString(mttr);
+                output.write(json.getBytes(encoding));
+
+                int statusCode = conn.getResponseCode();
+
+                try (InputStream in = (statusCode >= 200 && statusCode < 400)
+                        ? conn.getInputStream() : conn.getErrorStream();) {
+                    MTHUBTranslateResponse response = OM.readValue(
+                            IOUtils.toString(in, StandardCharsets.UTF_8),
+                            MTHUBTranslateResponse.class);
+                    String translation = "";
+                    if (response.isSuccess()) {
+                        translation = response.getData().
+                                      get(0).getSegments().get(0).get(0).
+                                        getTranslation();
+                        putToCache(sLang, tLang, text, translation);
+                    } else {
+                        switch (response.getError().get(0).code) {
+                            case 1: return OStrings.getString("MTHUB_ERROR",
+                                    response.getError().get(0).code,
+                                    OStrings.getString("MT_ENGINE_MTHUB_INVALID_KEY"));
+                            case 3: return OStrings.getString("MTHUB_ERROR",
+                                    response.getError().get(0).code,
+                                    OStrings.getString("MT_ENGINE_MTHUB_INVALID_LANG_CODE"));
+                            default: return OStrings.getString(
+                                        "MTHUB_ERROR", "400",
+                                        response.getError().get(0).getMessage());
+                        }
+                    }
+                    return translation;
+                } catch (Exception e) {
+                    return OStrings.getString("MTHUB_ERROR", "400", e.getMessage());
+                }
+            } catch (Exception e) {
+                return OStrings.getString("MTHUB_ERROR", "400", e.getMessage());
+            }
+        } catch (MalformedURLException e) {
+            return OStrings.getString("MTHUB_ERROR", "404", e.getMessage());
+        } catch (Exception e) {
+            return OStrings.getString("MTHUB_ERROR", "400", e.getMessage());
+        }
+    }
+
+    /**
+     * Normalise language codes
+     *
+     * @param language
+     *            An OmegaT language
+     * @return A normalise code for MT-HUB languages (ISO 639-1 Code)
+     */
+    private String normaliseCode(Language language) {
+
+        String fullCode = language.getLanguage().toUpperCase();
+
+        return availableLanguageCodes.contains(fullCode) ? fullCode : language.getLanguageCode();
```

Comment:
You should be using the getter to get the list of available codes; otherwise you risk an NPE.

Reply:
Can you change it by:
`
return this.getAvailableLanguageCodes().contains(fullCode) ? fullCode : language.getLanguageCode();`